docs/concepts/dependencies.md

```diff
@@ -223,6 +223,46 @@ members = [
 ]
 ```
 
+### Platform-specific sources
```

Comment:
The is one big catch we need to mention (and should check in code): The metadata needs to consistent between the sources, otherwise it's UB.

Reply:
I think different metadata is totally fine, no?

Reply:
Given:
```toml

        [tool.uv.sources]
        iniconfig = [
            { url = "https://files.pythonhosted.org/packages/ef/a6/62565a6e1cf69e10f5727360368e451d4b7f58beeac6173dc9db836a5b46/iniconfig-2.0.0-py3-none-any.whl", marker = "sys_platform != 'win32'" },
            { url = "https://files.pythonhosted.org/packages/d7/4b/cbd8e699e64a6f16ca3a8220661b5f83792b3017d0f79807cb8708d33913/iniconfig-2.0.0.tar.gz", marker = "sys_platform == 'win32'" },
        ]
```
do we check the metadata for both packages individually?

Reply:
Yeah. It’s just like if you had the same direct URL requirement defined twice, with disjoint markers.

Reply:
cool!